keepassxc-browser/background/browserAction.js

```diff
@@ -240,7 +248,39 @@ browserAction.setRememberPopup = function(tabId, username, password, url, userna
         browserAction.show(null, {'id': id});
 
         if (page.settings.showLoginNotifications) {
-            showNotification('Create or modify the credentials by clicking on the extension icon.');
+            const message = 'Create or modify the credentials by clicking on the extension icon.';
+            const buttons = [
+            {
+                'title': 'Close'
+            },
+            {
+                'title': 'Never ask for this page'
+            }];
+
+            browser.notifications.create({
+                'type': 'basic',
+                'iconUrl': browser.extension.getURL('icons/keepassxc_64x64.png'),
+                'title': 'KeePassXC-Browser',
+                'message': message,
+                'buttons': buttons
+            });
+
+            browser.notifications.onButtonClicked.addListener((id, index) => {
+                browser.notifications.clear(id);
+                if (index === 1) {
+                    browser.windows.getCurrent().then((win) => {
+                        browser.tabs.query({ 'active': true, 'currentWindow': true }).then((tabs) => {
+                            const tab = tabs[0];
+                            browser.runtime.getBackgroundPage().then((global) => {
+                                browser.tabs.sendMessage(tab.id, {
+                                    action: 'ignore-site',
+                                    args: [url]
+                                });
+                            });
+                        });
+                    });
+                }
+            });
```

Comment:
A bit deep, isn't it? Maybe see if you can reduce nesting or at least add a few comments so one can see which braces belong together.